stdlib/Printf/src/Printf.jl

```diff
@@ -77,18 +77,54 @@ end
 base(T) = T <: HexBases ? 16 : T <: Val{'o'} ? 8 : 10
 char(::Type{Val{c}}) where {c} = c
 
+struct InvalidFormatStringError <: Exception
+    message::String
+    format::String
+    start_color::Int
+    end_color::Int
+end
+
+function Base.showerror(io::IO, err::InvalidFormatStringError)
+    io_has_color = get(io, :color, false)
+
+    println(io, "InvalidFormatStringError: ", err.message)
+    print(io, "\t\"", @view(err.format[begin:prevind(err.format, err.start_color)]))
+    invalid_text = @view err.format[err.start_color:err.end_color]
+
+    if io_has_color
+        printstyled(io, invalid_text, color=:red)
+    else
+        print(io, invalid_text)
+    end
+
+    # +1 is okay, since all format characters are single bytes
+    println(io, @view(err.format[err.end_color+1:end]), "\"")
+
+    arrow_error = '-'^(length(invalid_text)-1)
+    arrow = "\t" * ' '^err.start_color * arrow_error * "^\n"
```

Comment:
Spaces instead of tab?

Reply:
yes

Reply:
I'll try some and see how many look good.